tests/reportArticleTest.js

```diff
@@ -0,0 +1,173 @@
+          expect(title).to.equal('Enough is Enough!');
+          expect(res.body.message).to.equal('You have created an article successfully');
+          const articleSlug = res.body.article.slug;
+          slug = articleSlug;
+          done();
+        });
+    });
+
+    it('Returns 401 error when user is not authenticated', (done) => {
+      chai
+        .request(app)
+        .post(`/api/articles/${slug}/report`)
+        .set({ authorization: 'userToken', Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .end((err, res) => {
+          expect(res.status).to.equal(401);
+          expect(res.body.message).to.equal('jwt malformed');
+          done();
+        });
+    });
+
+    it('Returns 404 error when article can not be found', (done) => {
+      chai
+        .request(app)
+        .post('/api/articles/slug/report')
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .end((err, res) => {
+          expect(res.status).to.equal(404);
+          expect(res.body.error).to.equal('Article Not found!');
+          done();
+        });
+    });
+
+    it('Should report an article successfully', (done) => {
+      chai
+        .request(app)
+        .post(`/api/articles/${slug}/report`)
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .send(article)
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          expect(res.body.message).to.equal('You have reported this article successfully');
+          done();
+        });
+    });
+
+    it('Should report an article successfully when field is not field', (done) => {
+      chai
+        .request(app)
+        .post(`/api/articles/${slug}/report`)
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .send({ violation: 'plagiarisism' })
+        .send(article)
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          expect(res.body.message).to.equal('You have reported this article successfully');
+          done();
+        });
+    });
+
+    it('Should return an error when token is invalid', (done) => {
+      chai
+        .request(app)
+        .get('/api/admin/reportedArticles')
+        .set({ authorization: 'invalidtoken', Accept: 'application/json' })
+        .end((err, res) => {
+          expect(res.status).to.equal(401);
+          expect(res.body.message).to.equal('jwt malformed');
+          done();
+        });
+    });
+
+    it('Should return error when user is not authorized', (done) => {
+      chai
+        .request(app)
+        .get('/api/admin/reportedArticles')
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .end((err, res) => {
+          expect(res.status).to.equal(403);
+          expect(res.body.message).to.equal('Access Denied!, You are not authorized to do this');
```

Comment:
I think this error message can be a bit more descriptive.

Reply:
Noted